src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,76 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+interface TodoModalProps {
+  onClose: () => void;
+  user: User | null;
+}
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+export const TodoModal: React.FC<TodoModalProps> = ({ onClose, user }) => {
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const [loading, setLoading] = useState(true);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (currentTodo) {
+        setLoading(false);
+      }
+    }, 300);
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+    return () => clearTimeout(timer);
+  }, [currentTodo]);
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+  if (!currentTodo) {
+    return null;
+  }
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+  return (
+    <div className="modal is-active" data-cy="modal">
+      <div className="modal-background" onClick={onClose} />
+      {loading ? (
+        <div className="loader-container">
+          <Loader />
+        </div>
+      ) : (
```

Comment:
Here you are conditionally rendering the loader or the modal content based on the `loading` state. This is a good practice. However, it might be more readable to move the modal content into a separate component. This way, the `TodoModal` component would not be responsible for both loading state and displaying the content.